interactions/api/events/processors/voice_events.py

```diff
@@ -13,15 +13,18 @@
 class VoiceEvents(EventMixinTemplate):
     @Processor.define()
     async def _on_raw_voice_state_update(self, event: "RawGatewayEvent") -> None:
-        before = copy.copy(self.cache.get_voice_state(event.data["user_id"])) or None
         after = await self.cache.place_voice_state_data(event.data)
+        is_bot = event.data["member"]["user"]["bot"]
 
-        self.dispatch(events.VoiceStateUpdate(before, after))
-
-        if before and before.user_id == self.user.id:
+        if is_bot:
             if vc := self.cache.get_bot_voice_state(event.data["guild_id"]):
                 # noinspection PyProtectedMember
-                await vc._voice_state_update(before, after, event.data)
+                await vc._voice_state_update(after, event.data)
+            return
+
+        # Following code is for updates to non-bot users
+        before = copy.copy(self.cache.get_voice_state(event.data["user_id"])) or None
+        self.dispatch(events.VoiceStateUpdate(before, after))
```

Comment:
We don't want to dispatch this voice state update for bots -- this would affect global state for the bot across all guilds.

Reply:
Um, we do though... people may listen to this update, and it's important bots get dispatched.

Reply:
@AstreaTSS Do you understand the problem though? This dispatch event is sending a before object for event data from the user ID. The user ID is not an individual user. It's the the bot itself. So this before data will always be completely wrong if the bot's being used in multiple guilds. I'm not really sure wat the use case is for the `VoiceStateUpdate`, but maybe just sending `self.cache.get_bot_voice_state(event.data["guild_id"])` as before data will work

```python
before = copy.copy(self.cache.get_voice_state(event.data["user_id"])) or None
```

Reply:
Well, the before attribute being the bot... makes sense? Unless I'm missing something, that seems perfectly reasonable.

Reply:
I think it might help if you look at my original issue I linked. The bot does not connect and disconnect from voice channels properly across guilds because this code is affecting and checking global voice state of the bot (`self.cache.get_voice_state(event.data["user_id"]`). There should be absolutely no global voice state for the bot, this makes no sense. There should be a voice state of the bot for each guild it's active in.

Reply:
Of course, that part makes sense... but this fix is completely wrong. The bot shouldn't *not* dispatch if it doesn't find a voice state, it should dispatch the specific state it's in.

Reply:
OK I'm sorry I was unaware of the purpose of the dispatch events, I haven't used those listeners before. I updated the `is_bot` block to dispatch the `VoiceUpdate` event, but now it's using the guild voice state as the before.